### DISCUSSION TITLE
Thaumcraft Tweaks

### DISCUSSION THREAD

**Discussion by Turkey9002** [ACGaming/UniversalTweaks#23] [Tweak Suggestions]
Noticed there was mod support too and I thought some love for Thaumcraft 6 would be nice. It's closed source unfortunately but you might be able to find a decompiled version somewhere (or decompile it yourself).

- [x] [Stable Thaumometer](https://www.curseforge.com/minecraft/mc-mods/stable-thaumometer)
Simply makes the Thaumometer not have a use (bob) animation while scanning with it.

- [ ] [Thaumic Inventory Scanning](https://www.curseforge.com/minecraft/mc-mods/thaumcraft-inventory-scanning)
Holding the Thaumometer over items in GUIs scans them. It doesn't work on Creative for some reason (at least whenever I used it myself) but it would be nice if that was fixed.

- [ ] [Thaumcraft 6 Enhanced](https://www.curseforge.com/minecraft/mc-mods/thaumcraft-6-enhanced) [?]
Modifies runic shielding to be more like previous Thaumcraft versions, fixes some [bugs](https://github.com/Sheenus/Thaumcraft6Enhanced/blob/main/src/main/java/thaumcraft/common/lib/events/PlayerEvents.java), and attempts to restore the runic shield hud. The mod itself isn't compatible with Thaumic Augmentation which also fixed some of Thaumcraft's worst bugs but it looks like it may or may not be maintained again soon.

**Bugfixes:**

- [x] Firebats do not emit particles unlike their previous Thaumcraft counterparts.
Not a tweak I know but I do know how to fix it. This can be fixed by injecting this into onLivingUpdate() in EntityFireBat:
```
if (this.world.isRemote) 
	{
            this.world.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, this.prevPosX + (this.world.rand.nextFloat() - this.world.rand.nextFloat()) * 0.2f, this.prevPosY + this.height / 2.0f + (this.world.rand.nextFloat() - this.world.rand.nextFloat()) * 0.2f, this.prevPosZ + (this.world.rand.nextFloat() - this.world.rand.nextFloat()) * 0.2f, 0.0D, 0.0D, 0.0D);
            this.world.spawnParticle(EnumParticleTypes.FLAME, this.prevPosX + (this.world.rand.nextFloat() - this.world.rand.nextFloat()) * 0.2f, this.prevPosY + this.height / 2.0f + (this.world.rand.nextFloat() - this.world.rand.nextFloat()) * 0.2f, this.prevPosZ + (this.world.rand.nextFloat() - this.world.rand.nextFloat()) * 0.2f, 0.0D, 0.0D, 0.0D);
        }
```

- [x] Wisp's death burst particle is too small compared to previous Thaumcraft versions.
Same as above and I also have the fix for this. Simply override onDeath() with this in EntityWisp:
```
public void onDeath(final DamageSource par1DamageSource) 
    {
        super.onDeath(par1DamageSource);
        if (this.world.isRemote) {
            FXDispatcher.INSTANCE.burst(this.posX, this.posY + 0.44999998807907104, this.posZ, 10.0f);
        }
    }
```
Thanks for this mod by the way. It's one of the biggest godsends of 1.12 modding. :)

**Comment by ACGaming** [comment]
Firebat particles implemented in https://github.com/ACGaming/UniversalTweaks/commit/04b0b8dc7ef5210767766a21c0870c23ddb2e69d
Wisp particles implemented in https://github.com/ACGaming/UniversalTweaks/commit/769b313931897fad4e222b0a8dd33557e2f68474

**Comment by ACGaming** [comment]
Stable thaumometer implemented in https://github.com/ACGaming/UniversalTweaks/commit/8456aefd16968dced5a59c9155e3813ece73d89d

**Comment by ACGaming** [comment]
Handled by Thaumcraft Fix or ThaumicTweaker.
